src/main/java/carpet/script/value/EntityValue.java

```diff
@@ -891,21 +893,51 @@ else if (v instanceof ListValue)
             throw new InternalExpressionException("'effect' needs either no arguments (clear) or effect name, duration, and optional amplifier, show particles and show icon");
         });
 
-        // gamemode
-        // spectate
-        // "fire"
-        // "extinguish"
-        // "silent"
-        // "gravity"
-        // "invulnerable"
-        // "dimension"
-        // "item"
-        // "count",
-        // "age",
-        // "effect_"name
-        // "hold"
-        // "hold_offhand"
-        // "jump"
+        put("gamemode_id", (e,v)->{
+            if(!(e instanceof ServerPlayerEntity)){
+                throw new InternalExpressionException("Cannot change the gamemode of non-player entities");
+            }
+            if(v.readInteger()==0){
```

Comment:
Also - readInterger is for scanf-like applications, check type first and then use NumericValue.asNumber(v).getLong()

also use switch statement or if - else if

Reply:
done